Sources/StytchUI/StytchB2BUIClient/ViewControllers/TOTPEntryViewController/TOTPEntryViewController.swift

```diff
@@ -56,7 +53,10 @@ final class TOTPEntryViewController: BaseViewController<TOTPEntryState, TOTPEntr
     }
 
     private func continueWithTOTP() {
-        navigationController?.pushViewController(RecoveryCodeSaveViewController(state: .init(configuration: viewModel.state.configuration)), animated: true)
+        // We only want to show the RecoveryCodeSaveViewController if we have a totp response with recovery codes
+        if B2BAuthenticationManager.totpResponse != nil {
+            navigationController?.pushViewController(RecoveryCodeSaveViewController(state: .init(configuration: viewModel.state.configuration)), animated: true)
+        }
     }
```

Comment:
This will prevent the recovery code save screen flicker from happening if we are not in enrollment.